packages/react-storage/src/components/StorageBrowser/views/LocationsView/LocationsView.tsx

```diff
@@ -99,34 +91,48 @@ export function LocationsView({
       hasError,
     });
 
-  return (
-    <div
-      className={resolveClassName(CLASS_BASE, className)}
-      data-testid="LOCATIONS_VIEW"
-    >
-      <Title>Home</Title>
-      <RefreshControl
-        disableRefresh={disableRefresh}
-        handleRefresh={() => {
+  // FIXME: Eventually comes from useView hook
+  const contextValue: ControlsContext = {
+    data: {},
+    actionsConfig: {
+      type: 'SINGLE_ACTION',
+      isCancelable: true, // Question(ashwinkumar6): is refresh Cancelable ?
```

Comment:
QQ
is this required for refresh 
```
 data: {},
    actionsConfig: {
      type: 'SINGLE_ACTION',
      isCancelable: true, // Q
```

Reply:
Left some previous comments [here](https://github.com/aws-amplify/amplify-ui/pull/5939/files#diff-9cdd03a53e36958be1ed87fb20255b4814ae5e41c45be51b0062b309144da0aeR26), and [here](https://github.com/aws-amplify/amplify-ui/pull/5939/files#diff-2a298ab664ed59a235a070ed3e4ca2e5045c4383dadd8afb5ceda79cc2c142a6R159) that are somewhat related here.

The `DataRefresh` should has to do with configuration - it just renders an icon button with some static data (a11y values, etc) and accepts a handful of `props` (`isDisabled`, `children` and `onRefresh`) so there is no reason that we need to add configuration related to `actions` here.

There maybe a piece to this approach that i am jus misunderstanding but regardless it is out of scope for adding the `DataRefresh` component

Reply:
Agree with @calebpollman here. That stuff is irrelevant on this view. It was only relevant for the StatusDisplay component because we were keying some conditional presentational logic off the action configuration. That shouldn't be applicable here

Reply:
discussed aync, updated to 
```
export interface ControlsContext {
  data: {
    taskCounts?: TaskCounts;
    isDataRefreshDisabled?: boolean; // <== data
  };
  actionsConfig: {
    type: 'SINGLE_ACTION' | 'BATCH_ACTION';
    isCancelable: boolean;
  },
  onDataRefresh?: () => void; // <== Handlers
}
```
With plans to refactor in upcoming PR